.travis.yml

```diff
@@ -0,0 +1,21 @@
+language: php
+
+sudo: false
+
+matrix:
+  include:
+    - php: 7.1
+    - php: 7.1
+      env: DEPENDENCIES='low'
+    - php: nightly
+  allow_failures:
+    - php: nightly
+  fast_finish: true
+
+install:
+  - composer self-update
+  - if [ "$DEPENDENCIES" != "low" ]; then composer update; fi;
+  - if [ "$DEPENDENCIES" == "low" ]; then composer update --prefer-lowest; fi;
```

Comment:
Could be simplified by setting an `env` var to `COMPOSER_FLAGS` and inject it automatically without any `if` statement, like this:

```yaml

env:
  COMPOSER_FLAGS: ''
matrix:
  include:
    - php: 7.1
    - php: 7.1
      env: COMPOSER_FLAGS='--prefer-lowest'
    - php: nightly
  allow_failures:
    - php: nightly
  fast_finish: true

install:
  - composer self-update
  - composer update $COMPOSER_FLAGS
```